### PR TITLE
assert_frame_equal for set

### LINKED ISSUE #51727: BUG: assert_frame_equal failed on comparing two not equal Sets ('set' object is not subscriptable)
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [ ] I have confirmed this bug exists on the [main branch](https://pandas.pydata.org/docs/dev/getting_started/install.html#installing-the-development-version-of-pandas) of pandas.

### Reproducible Example

```python
import pandas as pd
from pandas.testing import assert_frame_equal


df1 = pd.DataFrame({'set_column': [{1, 2, 3}, {4, 5, 6}]})  # last number is different.
df2 = pd.DataFrame({'set_column': [{1, 2, 3}, {4, 5, 7}]})

assert_frame_equal(df1, df2)
```

### Issue Description

Exception: 
```
pandas/_libs/testing.pyx:53: in pandas._libs.testing.assert_almost_equal
    ???
pandas/_libs/testing.pyx:159: in pandas._libs.testing.assert_almost_equal
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

>   ???
E   TypeError: 'set' object is not subscriptable

pandas/_libs/testing.pyx:159: TypeError
```

It was raised in [assert_series_equal](https://github.com/pandas-dev/pandas/blob/main/pandas/_testing/asserters.py#L1000) in last else statement:
```
_testing.assert_almost_equal(
            left._values,
            right._values,
            rtol=rtol,
            atol=atol,
            check_dtype=bool(check_dtype),
            obj=str(obj),
            index_values=np.asarray(left.index),
        )
```

### Expected Behavior

assert not equal error will be shown.

### Installed Versions

<details>

INSTALLED VERSIONS
------------------
commit           : 66e3805b8cabe977f40c05259cc3fcf7ead5687d
python           : 3.9.5.final.0
python-bits      : 64
OS               : Linux
OS-release       : 5.15.0-1028-aws
Version          : #32~20.04.1-Ubuntu SMP Mon Jan 9 18:02:08 UTC 2023
machine          : x86_64
processor        : x86_64
byteorder        : little
LC_ALL           : None
LANG             : C.UTF-8
LOCALE           : en_US.UTF-8
pandas           : 1.3.5
numpy            : 1.22.2
pytz             : 2022.7.1
dateutil         : 2.8.0
pip              : 22.3
setuptools       : 65.5.0
Cython           : 0.29.33
pytest           : 6.2.5
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : 3.0.8
lxml.etree       : 4.9.1
html5lib         : 1.1
pymysql          : None
psycopg2         : None
jinja2           : 3.0.1
IPython          : None
pandas_datareader: None
bs4              : None
bottleneck       : None
fsspec           : None
fastparquet      : None
gcsfs            : None
matplotlib       : 3.3.4
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : 3.0.0
pyxlsb           : None
s3fs             : None
scipy            : 1.5.4
sqlalchemy       : 1.4.46
tables           : None
tabulate         : None
xarray           : None
xlrd             : None
xlwt             : None
numba            : None

</details>

But also was able to reproduce on latest version `pandas==1.5.3`.

### PR DESCRIPTION
**Pull request by jayendra-patil33**
- [x] closes #51727 
- [x] [Tests added and passed]
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/tests/util/test_assert_almost_equal.py [hunk 1]
```diff
@@ -202,6 +202,16 @@ def test_assert_almost_equal_edge_case_ndarrays(left_dtype, right_dtype):
     )
 
 
+def test_assert_almost_equal_sets():
+    _assert_almost_equal_both({1, 2, 3}, {1, 2, 3})
```
Comment (topper-123) [inline pandas/tests/util/test_assert_almost_equal.py:206]: Can you add issue number here.

File: pandas/tests/util/test_assert_almost_equal.py [hunk 2]
```diff
@@ -202,6 +202,16 @@ def test_assert_almost_equal_edge_case_ndarrays(left_dtype, right_dtype):
...
+
+
+def test_assert_almost_not_equal_sets():
```
Comment (topper-123) [inline pandas/tests/util/test_assert_almost_equal.py:209]: Can you add issue number here also.

File: pandas/tests/util/test_assert_frame_equal.py [hunk 3]
```diff
@@ -362,3 +362,18 @@ def test_assert_frame_equal_ts_column():
     msg = r'DataFrame.iloc\[:, 0\] \(column name="a"\) values are different'
     with pytest.raises(AssertionError, match=msg):
         tm.assert_frame_equal(df1, df2)
+
+
+def test_assert_frame_equal_set():
+    df1 = DataFrame({"set_column": [{1, 2, 3}, {4, 5, 6}]})
```
Comment (topper-123) [inline pandas/tests/util/test_assert_frame_equal.py:368]: issue number

File: pandas/tests/util/test_assert_frame_equal.py [hunk 4]
```diff
@@ -362,3 +362,18 @@ def test_assert_frame_equal_ts_column():
...
+    df2 = DataFrame({"set_column": [{1, 2, 3}, {4, 5, 6}]})
+    tm.assert_frame_equal(df1, df2)
+
+
+def test_assert_frame_equal_set_mismatch():
+    df1 = DataFrame({"set_column": [{1, 2, 3}, {4, 5, 6}]})
```
Comment (topper-123) [inline pandas/tests/util/test_assert_frame_equal.py:374]: issue number.

**Review by topper-123 — COMMENTED** [review]
A few comments. Can you add line in `whatsnew/v2.1.0.rst` about this, under bug fixes/other.

File: doc/source/whatsnew/v2.1.0.rst [hunk 5]
```diff
@@ -244,7 +244,7 @@ Other
 
 .. ***DO NOT USE THIS SECTION***
 
--
+- Bug in :func:`assert_almost_equal` now throwing assertion error for two unequal sets (:issue:`51727`)
```
Comment (topper-123) [inline doc/source/whatsnew/v2.1.0.rst:247]: This must be straight under the `Other` section, above the ` ***DO NOT USE THIS SECTION***` section 😄 .

**Comment by topper-123** [comment]
Ping when ready and I'll merge it.

**Comment by jayendra-patil33** [comment]
@topper-123 done

**Comment by topper-123** [comment]
Thanks, @jayendra-patil33 .

**Pull request merged**